src/transactions/InvokeHostFunctionOpFrame.cpp

```diff
@@ -1041,7 +1027,7 @@ class InvokeHostFunctionParallelApplyHelper
     bool
     previouslyRestoredFromHotArchive(LedgerKey const& lk) override
     {
-        return mPreviouslyRestoredHotEntries.count(lk) > 0;
+        return mOpState.entryWasRestored(lk);
```

Comment:
nit: I think `previouslyRestoredFromHotArchive` function should be removed and the call should be inlined at the (only) call site. It generally doesn't make much sense, given that hot archive only exists past p23 and it would be very suspicious if we have called it in non-p23 code path.